### PR TITLE
Add CI structure and Build dependencies job

### PR DESCRIPTION
**Pull request by jparisu**
Signed-off-by: jparisu <javierparis@eprosima.com>

### CODE REVIEW

File: .github/actions/install-eprosima-dependencies-windows/action.yml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  cmake_build_type:
+    description: 'Specifies the build type on single-configuration generators'
+    required: true
+    default: 'Release'
+  fastdds_branch:
+    description: 'Branch or tag of Fast DDS repository (https://github.com/eProsima/Fast-DDS)'
+    required: true
+    default: 'master'
+  utils_branch:
+    description: 'Branch or tag of Dev Utils repository (https://github.com/eProsima/Fdev-utils)'
```
Comment (juanlofer-eprosima) [inline .github/actions/install-eprosima-dependencies-windows/action.yml:13]:
```suggestion
    description: 'Branch or tag of Dev Utils repository (https://github.com/eProsima/dev-utils)'
```

File: .github/actions/install-python-packages/action.yml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+          sphinxcontrib-imagehelper==1.1.1 \
+          sphinx-tabs==3.2.0 \
+          colcon-common-extensions \
+          colcon-mixin \
+          vcstool \
+          GitPython \
+          setuptools \
+          gcovr==5.0 \
+          pyyaml \
+          jsonschema
```
Comment (juanlofer-eprosima) [inline .github/actions/install-python-packages/action.yml:22]: NIT: We might want to reduce the dependencies to what it's actually required.

**Review by juanlofer-eprosima — CHANGES_REQUESTED** [review]

File: .github/actions/install-python-packages/action.yml (hunk 2 above)
Comment (jparisu) [reply .github/actions/install-python-packages/action.yml:22]: TODO

**Review by juanlofer-eprosima — APPROVED** [review]
LGTM

**Pull request merged**